### PR TITLE
docs: add docs for StorageImage

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Add documentation for `StorageImage`

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/connected-components/storage/storageimage/react.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+```js
+import { StorageImage } from '@aws-amplify/ui-react-storage';
+import '@aws-amplify/ui-react/styles.css';
+```
+</ExampleCode>
+
+At a minimum you must include the `alt`, `imgKey` and `accessLevel` props. `accessLevel` refers to the [Amplify Storage access level](https://docs.amplify.aws/lib/storage/configureaccess/q/platform/js/), which is `'public' | 'private' | 'protected'`.
+
+<Example>
+  <Image alt='cat' src='/cats/1.jpg' width="400px" height="400px" />
```
Comment (zchenwei) [inline docs/src/pages/[platform]/connected-components/storage/storageimage/react.mdx:39]: The existing storage mock is doing via a custom provider and it is not supported on `StorageImage` API. So simply use an `Image` to render the example.

**Review by zchenwei — COMMENTED** [review]
I try to deploy an app to hosting but now it requires admin access to our repo 😅

**Comment by hbuchel** [comment]
> I try to deploy an app to hosting but now it requires admin access to our repo 😅

@zchenwei I deployed it here https://chenwz-dev-imagedocs.dvmvffzts1tcu.amplifyapp.com/ (probably still building if you are reading this just after I posted this comment though)

**Comment by zchenwei** [comment]
> > I try to deploy an app to hosting but now it requires admin access to our repo 😅
> 
> @zchenwei I deployed it here https://chenwz-dev-imagedocs.dvmvffzts1tcu.amplifyapp.com/ (probably still building if you are reading this just after I posted this comment though)

Ohhh sweet! I ended up successfully deploying an app with hosting a few minutes ago. It is extremely hard to debug in hosting console with long logs 😭 <- https://chenwz-dev-imagedocs.daoba3qczgnpf.amplifyapp.com/react/connected-components/storage/storageimage

**Comment by hbuchel** [comment]
@zchenwei missing snapshot update for sitemap :) (my build failed on that)

**Review by thaddmt — APPROVED** [review]

File: docs/src/pages/[platform]/connected-components/storage/storageimage/react.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+<Alert variation="warning" heading="Wait!">
+  Did you follow the [quick start instructions](/connected-components/storage#quick-start) to set up the storage and auth services?
+</Alert>
+
+<Fragment platforms={['react']}>
+  {({ platform }) => import('@/components/AppDirectoryAlert')}
+</Fragment>
+
+To use the StorageImage component import it into your React application with the included styles.
```
Comment (reesscot) [inline docs/src/pages/[platform]/connected-components/storage/storageimage/react.mdx:25]:
```suggestion
To use the StorageImage component, import it into your React application with the included styles. 
```

File: docs/src/pages/[platform]/connected-components/storage/storageimage/props.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+export const STORAGE_IMAGE = [
+  {
+    name: 'imgKey',
```
Comment (reesscot) [inline docs/src/pages/[platform]/connected-components/storage/storageimage/props.ts:3]: Can we add alt here since it's required by underlying Image?

Comment (zchenwei) [reply docs/src/pages/[platform]/connected-components/storage/storageimage/props.ts:3]: Makes sense

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**